Resolve CI problems with jest-runner-eslint

For the last week or so `jest-runner-eslint` seems to be failing on all eslint versions being run against node 8 & 10, with this error:

```
FAIL lint src/rules/utils.ts
  ● Test suite failed to run

    Failed to load plugin 'eslint-plugin' declared in '../../home/runner/work/eslint-plugin-jest/eslint-plugin-jest/.eslintrc.js': Cannot find module 'eslint-plugin-eslint-plugin'
    Require stack:
    - /tmp/eslint-plugin-jest-xuJrTI/__placeholder__.js
    Referenced from: /home/runner/work/eslint-plugin-jest/eslint-plugin-jest/.eslintrc.js

      at Object.resolve (node_modules/eslint/lib/shared/relative-module-resolver.js:44:50)
          at _normalizeObjectConfigDataBody.next (<anonymous>)
          at _normalizeObjectConfigData.next (<anonymous>)
```

I've played around with CI a bit, and confirmed:

* Running `yarn lint` passes fine
* `ls`ing in places in `node_modules` shows that the `eslint-plugin-eslint-plugin` is there and contains the expected files & folders.
* Running locally works fine for me against Node 10.20.1
* The error happens using the previous 2 or 3 versions of `jest-runner-eslint`, so I'm pretty sure it's not because of a change in `jest-runner-eslint`

This seems to be happening on all PRs, so it doesn't seem to be because of a specific change. I also can't find any evidence of a change in actions, Node version, etc (i.e it doesn't seem like a bad version of Node 8 & 10 was released).

I've also noticed it seems like `Semantic Pull Request` has stopped working on PRs; CodeCov is also not showing, but I suspect that's just because jest is failing so it doesn't think all coverage has been collected.

I've hit a bit of a dead end in terms of things to investigate, so need a fresh set of eyes & creds.

PRs that are failing with this error include #585 & #588

--- a/src/rules/__tests__/no-deprecated-functions.test.ts
+++ b/src/rules/__tests__/no-deprecated-functions.test.ts
@@ -10,6 +10,11 @@ import rule, {
 
 const ruleTester = new TSESLint.RuleTester();
 
+// pin the original cwd so that we can restore it after each test
+const projectDir = process.cwd();
+
+afterEach(() => process.chdir(projectDir));
+
 /**
  * Makes a new temp directory, prefixed with `eslint-plugin-jest-`
  *

